src/plugins/index_pattern_field_editor/public/components/field_editor/form_schema.ts

```diff
@@ -43,17 +43,64 @@ export const schema: FormSchema<FieldFormInternal> = {
       label: i18n.translate('indexPatternFieldEditor.editor.form.defineFieldLabel', {
         defaultMessage: 'Define script',
       }),
+      validations: [
+        {
+          validator: emptyField(
+            i18n.translate(
+              'indexPatternFieldEditor.editor.form.validations.scriptIsRequiredErrorMessage',
+              {
+                defaultMessage: 'A script is required to set the field value.',
+              }
+            )
+          ),
+        },
+      ],
     },
   },
   customLabel: {
     label: i18n.translate('indexPatternFieldEditor.editor.form.customLabelLabel', {
       defaultMessage: 'Custom label',
     }),
+    validations: [
+      {
+        validator: emptyField(
+          i18n.translate(
+            'indexPatternFieldEditor.editor.form.validations.customLabelIsRequiredErrorMessage',
+            {
+              defaultMessage: 'Give a label to the field.',
+            }
+          )
+        ),
+      },
+    ],
   },
   popularity: {
     label: i18n.translate('indexPatternFieldEditor.editor.form.popularityLabel', {
       defaultMessage: 'Popularity',
     }),
+    validations: [
+      {
+        validator: emptyField(
+          i18n.translate(
+            'indexPatternFieldEditor.editor.form.validations.popularityIsRequiredErrorMessage',
+            {
+              defaultMessage: 'Give a popularity to the field.',
+            }
+          )
+        ),
+      },
+      {
+        validator: numberGreaterThanField({
+          than: 0,
+          message: i18n.translate(
+            'indexPatternFieldEditor.editor.form.validations.popularityGreaterThan0ErrorMessage',
+            {
+              defaultMessage: 'The popularity must be greater than zero.',
```

Comment:
Setting the field to zero is a valid use case even if its equivalent to hitting the field toggle. Lets allow it to be entered. 

`The popularity must be zero or greater.`

Reply:
Good point, thanks!